plugins/eventMacro/eventMacro.pl

```diff
@@ -196,157 +198,222 @@ sub commandHandler {
+						message( sprintf( "- orphan method: %s\n", $macro->orphan ), "list" );
+						message( sprintf( "- remaining repeats: %s\n", $macro->repeat ), "list" );
+						message( sprintf( "- slot: %s\n", $macro->slot ), "list" );
+						message( sprintf( "- macro delay: %s\n", $macro->macro_delay ), "list" );
+						
+						message( sprintf( "- current command: %s\n", $macro->{current_line} ), "list" );
+						
+						my $time_until_next_command = (($macro->timeout->{time} + $macro->timeout->{timeout}) - time);
+						message( sprintf( "- time until next command: %s\n", $time_until_next_command ), "list" ) if ($time_until_next_command > 0);
+						
+						message "\n";
+						
+						last if (!defined $macro->{subcall});
+						$macro = $macro->{subcall};
+					}
 				}
-			} else {
-				message "There's no macro currently running.\n";
-			}
-		}
-		if (!defined $params[0] || $params[0] eq 'automacro') {
-			my $status = $eventMacro->get_automacro_checking_status();
-			if ($status == CHECKING_AUTOMACROS) {
-				message "Automacros are being checked normally.\n";
-			} elsif ($status == PAUSED_BY_EXCLUSIVE_MACRO) {
-				message "Automacros are not being checked because there's an uninterruptible macro running ('".$eventMacro->{Macro_Runner}->last_subcall_name."').\n";
-			} elsif ($status == PAUSE_FORCED_BY_USER) {
-				message "Automacros checking is stopped because the user forced it.\n";
-			} else {
-				message "Automacros checking is active because the user forced it.\n";
+				message "\n";
 			}
 		}
 		
 	### parameter: check
 	} elsif ($arg eq 'check') {
-		if (!defined $params[0] || (defined $params[0] && $params[0] ne 'force_stop' && $params[0] ne 'force_start' && $params[0] ne 'resume')) {
-			message "usage: eventMacro check [force_stop|force_start|resume]\n", "list";
-			message 
-				"eventMacro check force_stop: forces the stop of automacros checking\n".
-				"eventMacro check force_start: forces the start of automacros checking\n".
-				"eventMacro check resume: return automacros checking to the normal state\n";
-			return;
+        if (!defined $params[0] || (defined $params[0] && $params[0] ne 'force_stop' && $params[0] ne 'force_start' && $params[0] ne 'resume')) {
+            message "usage: eventMacro check [force_stop|force_start|resume]\n", "list";
+            message
+                "eventMacro check force_stop [slot]: forces the stop of automacros checking [in a certain slot].\n".
+                "eventMacro check force_start [slot]: forces the start of automacros checking [in a certain slot].\n".
+                "eventMacro check resume [slot]: return automacros checking to the normal state [in a certain slot].\n";
+            return;
+        }
+		
+		my $new_status = $params[0];
+       
+        if (defined $params[1]) {
+            my $slot = $params[1];
+			
+            if ($slot !~ /^\d+$/) {
+                message "[eventMacro] '".$slot."' is not a valid option\n";
```

Comment:
Should we allow people to target macros by the name of the running routine? So slot could be `foo` if the running macro's name is `foo`. Targeting by slots could be hard, since slots are allocated automatically.

Reply:
slots are not allocated automatically, they are defined in the automacro or in the plugin command.